validate/validate.go

```diff
@@ -0,0 +1,163 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/kong/deck/state"
+	"github.com/kong/deck/utils"
+	"github.com/kong/go-kong/kong"
+)
+
+type Validator struct {
+	ctx               context.Context
+	state             *state.KongState
+	client            *kong.Client
+	parallelism       int
+	rbacResourcesOnly bool
+}
+
+func NewValidator(
+	ctx context.Context,
+	ks *state.KongState,
+	client *kong.Client,
+	parallelism int,
+	rbacResourcesOnly bool,
+) *Validator {
+	return &Validator{
+		ctx:               ctx,
+		state:             ks,
+		client:            client,
+		parallelism:       parallelism,
+		rbacResourcesOnly: rbacResourcesOnly,
+	}
+}
+
+type ErrorsWrapper struct {
+	Errors []error
+}
+
+func (v ErrorsWrapper) Error() string {
+	var errStr string
+	for _, e := range v.Errors {
+		errStr += fmt.Sprintf("%s\n", e.Error())
```

Comment:
Do not add a print after the last error in the array.

Reply:
Done.